telicent_lib/adapter.py

```diff
@@ -67,17 +68,19 @@ def aborted(self) -> None:
             self.data_catalog_sink.close()
         super().aborted()
 
-    def update_data_catalog(self, headers: list[tuple[str, str | bytes | None]] | None = None) -> bool:
+    def update_data_catalog(self, headers: Mapping = None) -> bool:
         if not self.has_data_catalog:
             logger.warning("Cannot create data catalogue update as 'has_data_catalogue' is False")
             return False
-        payload = {
-            "data-source-name": self.source_name,
-            "data-source-type": self.source_type,
-            "data-source-last-update": datetime.datetime.now().astimezone().isoformat(),
-            "component-name": self.name
-        }
-        record = Record(headers=headers, key=None, value=json.dumps(payload), raw=None)
+        record = self.dataset.update_record(headers)
+        self.data_catalog_sink.send(record)
+        return True
+
+    def register_data_catalog(self, registration_fields: Mapping, headers: Mapping = None) -> bool:
+        if not self.has_data_catalog:
+            logger.warning("Cannot create data catalogue update as 'has_data_catalogue' is False")
+            return False
```

Comment:
Could be extracted to decorator to avoid repetition plus it will clean the code a bit:
```python
def require_data_catalog(method):
    @functools.wraps(method)
    def wrapper(self, *args, **kwargs):
        if not self.has_data_catalog:
            logger.warning("Cannot proceed as 'has_data_catalog' is False")
            return False
        return method(self, *args, **kwargs)
    return wrapper
 
 @require_data_catalog   
 def register_data_catalog(self, registration_fields: Mapping, headers: Mapping = None) -> bool:
```

This is a suggestion not a change request.